napari_pyclesperanto_assistant/_pipeline.py

```diff
@@ -60,18 +91,54 @@ class Pipeline:
 
     def _generate(self, vistor):
         yield vistor.header()
+        yield vistor.imports()
         for n, step in enumerate(self.steps):
+            yield vistor.subheader(step, n)
             yield vistor.operate(step, n)
             if self.show:
                 yield vistor.show(step, n)
-            yield ""  # newline
+            #yield ""  # newline
 
     def to_jython(self, filename=None):
         code = "\n".join(self._generate(JythonGenerator))
         if filename:
             Path(filename).write_text(code)
         return code
 
+    def to_notebook(self, filename=None):
+        # Todo: I assume there is a better way of doing the following 3 lines
```

Comment:
@tlambert03 I was struggling below to make a `cells` list that doesn't throw error messages further down. I thought something simple like `cells = [self._generate(NotebookGenerator)]` does the job, but there were weird errors happening afterwards. The code here works but any hint on making it better is welcome :-)